### PR TITLE
Build system updates and minor fixes

### PR DESCRIPTION
**Pull request by amadio**

### CODE REVIEW

File: cmake/modules/RootBuildOptions.cmake [hunk 1]
```diff
@@ -143,11 +143,7 @@ ROOT_BUILD_OPTION(opengl ON "Enable support for OpenGL (requires libGL and libGL
 ROOT_BUILD_OPTION(oracle ON "Enable support for Oracle databases (requires Oracle Instant Client)")
 ROOT_BUILD_OPTION(pgsql ON "Enable support for PostgreSQL")
 ROOT_BUILD_OPTION(pyroot ON "Enable support for automatic Python bindings (PyROOT)")
-if(WIN32)
-  ROOT_BUILD_OPTION(pyroot_experimental OFF "Use experimental Python bindings for ROOT")
-else()
-  ROOT_BUILD_OPTION(pyroot_experimental ON "Use experimental Python bindings for ROOT")
-endif()
+ROOT_BUILD_OPTION(pyroot_experimental ON "Use experimental Python bindings for ROOT")
```
Comment (Axel-Naumann) [inline cmake/modules/RootBuildOptions.cmake]: @bellenot is this what we want?

Comment (amadio) [reply cmake/modules/RootBuildOptions.cmake]: No, see the full commit. The way to adjust platform defaults is to turn it off below.

**Comment by amadio** [comment]
About cuDNN, it's not respecting fail-on-missing. We have to fix that. The default should be the same as CUDA in my opinion, but as it requires an nvidia account to download and install, I strongly suggest to have it off by default. @Axel-Naumann, what do you think?

**Comment by lmoneta** [comment]
I am not saying that the system should fail if missing, but if cuda is enabled and you have already cudnn in the system, by default it should be enabled !

**Comment by amadio** [comment]
We have a `fail-on-missing` option. When that's enabled, it should fail if missing. We can enable it with `all=ON`, though. I think that now that several people made changes to the options, it's out of sync again with the available options.

**Comment by amadio** [comment]
I've removed the cuDNN commit. I will send another PR for that later.

**Comment by amadio** [comment]
Jenkins is stuck waiting for a 32bit node to build on. The Windows build failed because a lock file already exists, so it failed to clone. All other builds passed.

**Comment by bellenot** [comment]
I fixed the Windows node

**Pull request merged**